homeassistant/data_entry_flow.py

```diff
@@ -256,6 +256,20 @@ def async_has_matching_flow(
                 return True
         return False
 
+    @callback
+    def async_has_matching_flow(
+        self, flow: FlowHandler[_FlowResultT, _HandlerT]
+    ) -> bool:
+        """Check if an existing matching flow is in progress."""
+        if not (flows := self._handler_progress_index.get(flow.handler)):
+            return False
+        for other_flow in flows:
+            if other_flow is flow:
+                continue
+            if flow.is_matching(other_flow):
```

Comment:
```suggestion
            if other_flow is not flow and flow.is_matching(other_flow):
```
minor nit: could be combined